lib/src/create_report.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import '../models.dart';
+import 'pubspec.dart';
+
+Future<Report> createReport(
+    String packageDir, ToolEnvironment toolEnvironment) async {
+  Pubspec pubspec;
+  try {
+    pubspec = Pubspec.parseFromDir(packageDir);
+  } on Exception catch (e) {
+    throw Exception('Cannot not create report without a pubspec.yaml: $e');
+  }
+
+  return Report(sections: [
+    _supportsDart2(pubspec),
+    // TODO(sigurdm):Implement rest of sections.
+  ]);
+}
+
+ReportSection _supportsDart2(Pubspec pubspec) {
+  final supportsDart2 = pubspec.supportsDart2;
+  final issues = <_Issue>[];
+
+  if (!supportsDart2) {
+    final environment = pubspec.toJson()['environment'] as Map;
+    SourceSpan span;
+    if (environment is YamlMap) {
+      final sdk = environment.nodes['sdk'];
+      if (sdk is YamlNode) {
+        span = sdk.span;
+      }
+    }
+
+    issues.add(
+      _Issue(
+        'The current sdk constraint does not allow any Dart 2 versions',
+        span: span,
+      ),
+    );
+  }
+
+  return ReportSection(
+      title: 'Package supports Dart 2',
+      maxPoints: 20,
+      grantedPoints: supportsDart2 ? 20 : 0,
+      summary: _makeSummary(
+          'Package gets 20 points if their dart sdk constraint allows Dart 2',
```

Comment:
In this case wouldn't it be fair to make two kinds of messages depending on whether or not Dart 2 is supported?

So:
```
Package gets 20 points because the Dart SDK constraint `$sdk` is compatible with Dart 2.
```
or 
```
Package gets 0 points because the Dart SDK constraint `$sdk` is not compatible with Dart 2.
```
or
```
Package gets 0 points because it does not feature an SDK constraint and is not considered Dart 2 compatible.
```

Reply:
Done

Reply:
I think the summary headline should always be the same, and then the details below explain what you did wrong